jwst/associations/tests/helpers.py

```diff
@@ -174,41 +170,57 @@ def combine_pools(pools, **pool_kwargs):
     global_env = globals()
     for row in mega_pool:
         mega_pool[row.index] = [
-            parse_value(v, global_env=global_env, local_env=local_env)
-            for v in row
+            parse_value(v, global_env=global_env, local_env=local_env) for v in row
         ]
 
     return mega_pool
 
 
 def parse_value(v, global_env=None, local_env=None):
-    """Evaluate if indicated"""
+    """
+    Evaluate if indicated.
+
+    Parameters
+    ----------
+    v : str
+        The string to be parsed.
+    global_env : dict
+        The classes defined in the globals environment.
+    local_env : dict
+        The classes defined in the locals environment.
+
+    Returns
+    -------
+    str
+        The parsed string value.
+    """
     if global_env is None:
         global_env = globals()
     if local_env is None:
         local_env = locals()
 
     result = v
     try:
-        m = re.match('@!(.+)', v)
+        m = re.match("@!(.+)", v)
     except TypeError:
         pass
     else:
         if m:
-            result = eval(m.group(1), global_env, local_env)
+            result = eval(m.group(1), global_env, local_env)  # noqa: S307
```

Comment:
was something wrong with the Ruff-suggested replacement `ast.literal_eval()` here?

Reply:
Yes, this usage requires the presence of the two environments, which `ast.literal_eval` does not allow. This should be removed through a testing refactor, but is probably beyond the scope of this PR.